src/main/java/seedu/address/model/house/UniqueHouseList.java

```diff
@@ -0,0 +1,166 @@
+    public void add(House toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateHouseException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /* WE KEEP THIS HERE IN CASE GOT EDITHOUSE YA
+    /**
+     * Replaces the person {@code target} in the list with {@code editedPerson}.
+     * {@code target} must exist in the list.
+     * The person identity of {@code editedPerson} must not be the same as another existing person in the list.
+
+    public void setPerson(Person target, Person editedPerson) {
+        requireAllNonNull(target, editedPerson);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new PersonNotFoundException();
+        }
+
+        if (!target.isSamePerson(editedPerson) && contains(editedPerson)) {
+            throw new DuplicatePersonException();
+        }
+
+        internalList.set(index, editedPerson);
+    }
+    */
+
+
+
+    /**
+     * Removes the equivalent house from the list.
+     * The house must exist in the list.
+     */
+    public void remove(House toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new HouseNotFoundException();
+        }
+    }
+
+    /**
+     * Replaces the contents of this list with {@code houses}.
+     * {@code houses} must not contain duplicate houses.
+     */
+    public void setHouses(List<House> houses) {
+        requireAllNonNull(houses);
+        if (!housesAreUnique(houses)) {
+            throw new DuplicateHouseException();
+        }
+
+        internalList.setAll(houses);
+    }
+
+    /**
+     * Returns the backing list as an unmodifiable {@code ObservableList}.
+     */
+    public ObservableList<House> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    @Override
+    public Iterator<House> iterator() {
+        return internalList.iterator();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof UniqueHouseList)) {
+            return false;
+        }
+
+        UniqueHouseList otherUniqueHouseList = (UniqueHouseList) other;
+        return internalList.equals(otherUniqueHouseList.internalList);
+    }
+
+    @Override
+    public int hashCode() {
+        return internalList.hashCode();
+    }
+
+    @Override
+    public String toString() {
+        return internalList.toString();
+    }
+
+    /**
+     * Returns true if {@code houses} contains only unique houses.
+     */
+    private boolean housesAreUnique(List<House> houses) {
+        for (int i = 0; i < houses.size() - 1; i++) {
+            for (int j = i + 1; j < houses.size(); j++) {
+                if (houses.get(i).equals(houses.get(j))) {
+                    return false;
+                }
+            }
+        }
+        return true;
```

Comment:
Might be better to implement duplicate check this way?
```
Set<House> set = HashSet<House>(houses);
if(set.size() < houses.size()) {
    return true;
}

return false;
```

Reply:
this way also can:

private boolean housesAreUnique(List<House> houses) {
    Set<House> houseSet = new HashSet<>(houses);
    return houseSet.size() == houses.size();
}